app/signals/apps/services/domain/signals/delete.py

```diff
@@ -0,0 +1,118 @@
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (C) 2023 Gemeente Amsterdam
+import logging
+import uuid
+
+from django.conf import settings
+from django.core.files.storage import default_storage
+from django.db import transaction
+from django.utils import timezone
+
+from signals.apps.search.tasks import delete_from_elastic
+from signals.apps.signals.models import DeletedSignal, Signal
+from signals.apps.signals.workflow import AFGEHANDELD, GEANNULEERD, GESPLITST
+from signals.celery import app
+
+log = logging.getLogger(__name__)
+
+
+class DeleteSignalsService:
+    """
+    A service class for deleting signals in a specific state after a certain period of time.
+
+    This class provides methods for deleting signals that meet the specified criteria.
+    """
+    valid_states = [AFGEHANDELD, GEANNULEERD, GESPLITST, ]
+
+    @staticmethod
+    def run(state: str, days: int, delay_deletion: bool = False, dry_run: bool = False) -> uuid.UUID:
+        """
+        Run the deletion process for signals.
+
+        Args:
+            state (str): The state of signals to be deleted.
+            days (int): The minimum number of days a signal should have been in the state to be eligible for deletion.
+            delay_deletion (bool, optional): If True, delay the actual deletion process using celery.
+            dry_run (bool, optional): If True, no actual deletion occurs.
+
+        Raises:
+            ValueError: If the provided state, days, or feature flag is invalid.
+        """
+        if not settings.FEATURE_FLAGS.get('DELETE_SIGNALS_IN_STATE_X_AFTER_PERIOD_Y_ENABLED', False):
+            raise ValueError('Feature flag "DELETE_SIGNALS_IN_STATE_X_AFTER_PERIOD_Y_ENABLED" is not enabled')
+
+        if state not in DeleteSignalsService.valid_states:
+            raise ValueError('Invalid state(s) provided must be one of '
+                             f'"{", ".join(DeleteSignalsService.valid_states)}"')
+
+        if days < 365:
+            raise ValueError('Invalid days provided must be at least 365')
+
+        batch_uuid = uuid.uuid4()
+
+        signal_qs = Signal.objects.filter(
+            parent_id__isnull=True,
+            status__state=state,
+            status__created_at__lt=timezone.now() - timezone.timedelta(days=days)
+        )
+
+        for signal in signal_qs:
+            if delay_deletion:
+                DeleteSignalsService.delete_signal.delay(signal.id, batch_uuid, dry_run)
+            else:
+                DeleteSignalsService.delete_signal(signal.id, batch_uuid, dry_run)
+
+        return batch_uuid
+
+    @staticmethod
+    @app.task(prio=0)
+    def delete_signal(signal_id: int, batch_uuid: uuid.UUID = None, dry_run: bool = False):  # noqa C901
```

Comment:
Perhaps this could be separated to adhere to the single responsibility principle?